config.splitting.js

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 Algolia Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This config can split long content into more than 1 record per resource.
+module.exports = {
+  actions: [
+    {
+      recordExtractor: ({ $, url }) => {
+        // Settings and other constants
+
+        const MAX_RECORD_LENGTH = 10000; // expressed in number of utf-8 characters
+        const TEXT_ELEMENTS = ['p', 'li'];
+        const WORD_SEPARATOR = ' '; // character to separate words in records
+
+        // Page metadata
+
+        const pageMeta = {
+          title: $('title').text(),
+          description: $('meta[name="description"]').attr('content'),
+        };
+
+        // Content Extraction helpers
+
+        const turnEmptyTagsToSeparators = $elem =>
+          $elem.children().each((i, child) => {
+            if (!$(child).text()) $(child).text(WORD_SEPARATOR);
+          });
+
+        const forEachTextElement = fct =>
+          $(TEXT_ELEMENTS.join(', ')).each((i, elem) => {
+            const $elem = $(elem);
+            turnEmptyTagsToSeparators($elem);
+            fct({
+              text: $elem
+                .text()
+                .trim()
+                .replace(/\s+/g, WORD_SEPARATOR), // de-duplicate whitespace (i.e. space or line break) into word separators
+            });
+          });
+
+        // Indexing helpers
+
+        const createRecord = ({ text = '', part }) => ({
+          objectID: `${url.pathname} ${part}`,
+          path: url.pathname.split('/'),
+          ...pageMeta,
+          text,
+        });
+
+        const recordsAccu = (() => {
```

Comment:
I find that overkill for what can be replaced in the end by:
```js
const records = [];
...
records.push({ ...baseRecord, text });
...
return records;
```
We should keep those examples as simple as possible for them to be accessible to everyone.

Reply:
I had done it that way to introduce more semantic in the code, e.g.  `recordsAccu.getNextIndex()` instead of just passing `records.length` as parameter of `createRecord()` without understanding why.

That said, I'm happy to revert to the logic we had initially, as you're suggesting, if you find it more legible.

Reply:
done.

Reply:
I find it quite clear with the comment that you've added in front of it. But that's only my opinion, others may prefer the former syntax, we can ask them.